crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs

```diff
@@ -361,26 +361,36 @@ impl<'a> ExtensionLogic for ForgeExtension<'a> {
                 let curve: Felt = input_reader.read()?;
                 let curve = curve.to_short_string().ok();
 
-                let (signing_key_bytes, verifying_key_bytes) = {
+                let (signing_key_bytes, x_coordinate_bytes, y_coordinate_bytes) = {
                     match curve.as_deref() {
                         Some("Secp256k1") => {
                             let signing_key = k256::ecdsa::SigningKey::random(
                                 &mut k256::elliptic_curve::rand_core::OsRng,
                             );
-                            let verifying_key = signing_key.verifying_key();
+                            let verifying_key = signing_key
+                                .verifying_key()
+                                .to_encoded_point(false)
+                                .to_bytes();
+                            let verifying_key = verifying_key.iter().as_slice();
                             (
-                                signing_key.to_bytes(),
-                                verifying_key.to_encoded_point(false).to_bytes(),
+                                signing_key.to_bytes().as_slice()[0..32].try_into().unwrap(),
+                                verifying_key[1..33].try_into().unwrap(),
+                                verifying_key[33..65].try_into().unwrap(),
```

Comment:
Let's wrap this into a small helper function since we are calling it in two different places. It can maybe even be an inline closure in this match arm.

Reply:
Updated! @cptartur